specs/dispute-game-interface.md

```diff
@@ -24,53 +25,188 @@ dispute games to exist. If multiple dispute games run in production, it gives
 
-Any contract that can be disputed by a dispute game implementation should
-implement the disputable interface. This allows for any dispute game to
-successfully interact with a variety of contracts.
+    /// @notice Returns the address of the `BondManager` used 
+    function bondManager() public view returns (IBondManager _bondManager);
 
-```solidity
-/// @title IDisputable
-/// @notice The generic interface for a disputable contract
-interface IDisputable {
-    /// @notice The implementation of the DisputeGame calls this after a
-    /// successful dispute. This should be guarded such that it is only
-    /// callable by a DisputeGame.
-    function challenge(bytes memory _data) external;
+    /// @notice If all necessary information has been gathered, this function should mark the game
+    ///         status as either `CHALLENGER_WINS` or `DEFENDER_WINS` and return the status of
+    ///         the resolved game. It is at this stage that the bonds should be awarded to the
+    ///         necessary parties.
+    /// @dev May only be called if the `status` is `IN_PROGRESS`.
+    function resolve() public returns (GameStatus _status);
+}
+
+////////////////////////////////////////////////////////////////
+//              OUTPUT ATTESTATION DISPUTE GAME               //
+////////////////////////////////////////////////////////////////
+
+/// @title IDisputeGame_OutputAttestation
+/// @notice The interface for an attestation-based DisputeGame meant to contest output
+///         proposals in Optimism's `L2OutputOracle` contract.
+interface IDisputeGame_OutputAttestation is IDisputeGame {
+    /// @notice A mapping of addresses from the `signerSet` to booleans signifying whether
+    ///         or not they have authorized the `rootClaim` to be invalidated.
+    function challenges(address challenger) external view returns (bool _challenged);
+
+    /// @notice Returns the list of authorized public keys that may challenge the `rootClaim`.
+    /// @return An array of authorized signers that may challenge the `rootClaim`.
+    function signerSet() external view returns (address[] memory _signers);
+
+    /// @notice The amount of signatures required to successfully challenge the `rootClaim`
+    ///         output proposal. Once this threshold is met by members of the `signerSet`
+    ///         calling `challenge`, the game will be resolved to `CHALLENGER_WINS`.
+    /// @custom:invariant The `signatureThreshold` may never be greater than the length of the `signerSet`.
+    function signatureThreshold() public view returns (uint16 _signatureThreshold);
+
+    /// @notice Challenge the `rootClaim`.
+    /// @dev - If the `ecrecover`ed address that created the signature is not a part of the
+    ///      signer set returned by `signerSet`, this function will revert.
+    ///      - If the `ecrecover`ed address that created the signature is not the msg.sender,
+    ///      this function will revert.
```

Comment:
Since this is an interface, what do you think about just saying `should` as opposed to `will`?

I only lean towards `should` since maybe there will be other schemes that unpack the signature differently (?).